File: legacy/_1.10.2/src/main/java/cn/enaium/noexpensive/NoExpensive.java
package cn.enaium.noexpensive;

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback;
import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback;
import cn.enaium.noexpensive.callback.EnchantmentCanCombineCallback;
import cn.enaium.noexpensive.callback.impl.AnvilSetOutputCallbackImpl;
import cn.enaium.noexpensive.callback.impl.AnvilTakeOutputCallbackImpl;
import cn.enaium.noexpensive.callback.impl.EnchantmentCanCombineCallbackImpl;
import cn.enaium.noexpensive.command.NoExpensiveCommand;
import net.fabricmc.api.ModInitializer;
import net.legacyfabric.fabric.api.registry.CommandRegistry;

/**
 * @author Enaium
 */
public class NoExpensive implements ModInitializer {
    @Override
    public void onInitialize() {
        System.out.println("Hello NoExpensive world!");

        CommandRegistry.INSTANCE.register(new NoExpensiveCommand());

        EnchantmentCanCombineCallback.EVENT.register(new EnchantmentCanCombineCallbackImpl());
        AnvilSetOutputCallback.EVENT.register(new AnvilSetOutputCallbackImpl());
        AnvilTakeOutputCallback.EVENT.register(new AnvilTakeOutputCallbackImpl());

        Config.load();
        Runtime.getRuntime().addShutdownHook(new Thread(Config::save));
    }
}

File: legacy/_1.11.2/src/main/kotlin/cn/enaium/noexpensive/NoExpensive.kt
package cn.enaium.noexpensive

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback
import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback
import cn.enaium.noexpensive.callback.EnchantmentCanCombineCallback
import cn.enaium.noexpensive.callback.impl.AnvilSetOutputCallbackImpl
import cn.enaium.noexpensive.callback.impl.AnvilTakeOutputCallbackImpl
import cn.enaium.noexpensive.callback.impl.EnchantmentCanCombineCallbackImpl
import cn.enaium.noexpensive.command.NoExpensiveCommand
import net.legacyfabric.fabric.api.registry.CommandRegistry

/**
 * @author Enaium
 */
fun initializer() {
    println("Hello NoExpensive world!")
    CommandRegistry.INSTANCE.register(NoExpensiveCommand())
    EnchantmentCanCombineCallback.EVENT.register(EnchantmentCanCombineCallbackImpl())
    AnvilSetOutputCallback.EVENT.register(AnvilSetOutputCallbackImpl())
    AnvilTakeOutputCallback.EVENT.register(AnvilTakeOutputCallbackImpl())
    Config.load()
    Runtime.getRuntime().addShutdownHook(Thread { Config.save() })
}
